karapace/schema_reader.py

```diff
@@ -208,7 +209,8 @@ def handle_msg(self, key, value):
                 }
                 self.log.info("Setting schema_id: %r with schema: %r", value["id"], value["schema"])
                 self.schemas[value["id"]] = value["schema"]
-                self.global_schema_id = value["id"]
+                if value["id"] > self.global_schema_id:  # Not an existing schema
+                    self.global_schema_id = value["id"]
```

Comment:
i was wondering actually if it was actually worth wrapping all update code to the global id inside a method , and locking on that, since we're still doing increments in the main thread, and comparisons + updates here

Reply:
I think that would be a good idea but can be done in a separate PR.